verifiers/internal/gha/npm_sigstore_tuf.go

```
@@ -0,0 +1,114 @@
+package gha
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"path"
+	"runtime"
+	"time"
+
+	sigstoreTuf "github.com/sigstore/sigstore-go/pkg/tuf"
+)
+
+const (
+	AttestationKeyUsage = "npm:attestations"
+	AttestationKeyID    = "SHA256:jl3bwswu80PjjokCgh0o2w5c2U4LhQAE57gj9cz1kzA"
+	TargetPath          = "registry.npmjs.org/keys.json"
+)
+
+// NpmjsKeysTarget describes the structure of the target file.
+type NpmjsKeysTarget struct {
+	Keys []Key `json:"keys"`
+}
+type Key struct {
+	KeyID     string    `json:"keyId"`
+	KeyUsage  string    `json:"keyUsage"`
+	PublicKey PublicKey `json:"publicKey"`
+}
+type PublicKey struct {
+	RawBytes   string   `json:"rawBytes"`
+	KeyDetails string   `json:"keyDetails"`
+	ValidFor   ValidFor `json:"validFor"`
+}
+type ValidFor struct {
+	Start time.Time `json:"start"`
+}
+
+type SigstoreTufClient interface {
+	GetTarget(target string) ([]byte, error)
+}
+
+// NewSigstoreTufClient Get a Sigstore TUF client, which itself is a wrapper around the official TUF client.
+func NewSigstoreTufClient() (*sigstoreTuf.Client, error) {
+	_, filename, _, ok := runtime.Caller(1)
+	if !ok {
+		log.Fatal("unable to get path")
```

return an error instead of logging